presentation/testing/src/test/IModelUtilities.test.ts

```diff
@@ -2,173 +2,184 @@
-    expect(unlinkFake.calledOnce);
-  });
+    it("does not call IModelJsFs.mkdirSync if output file exists", async () => {
+      const fileName = "fileName";
+      setupSnapshot();
+      sinon.stub(IModelJsFs, "existsSync").returns(true);
+      const mkdirFake = sinon.fake();
+      sinon.replace(IModelJsFs, "mkdirSync", mkdirFake);
+      const unlinkFake = sinon.fake();
+      sinon.replace(IModelJsFs, "unlinkSync", unlinkFake);
 
-  it("calls SnapshotDb.createEmpty with correct parameters", async () => {
-    const fileName = "fileName";
-    snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
-    const fake = sinon.fake.returns(snapshotDbMock.object);
-    sinon.replace(SnapshotDb, "createEmpty", fake);
-    sinon.stub(SnapshotConnection, "openFile");
+      await buildTestIModel(fileName, () => { });
 
-    await buildTestIModel(fileName, () => {});
+      expect(mkdirFake.notCalled);
+    });
 
-    expect(fake.firstCall.firstArg).to.include(`${fileName}.bim`);
-    expect(fake.firstCall.lastArg).to.deep.equal({ rootSubject: { name: fileName } });
-  });
+    it("calls SnapshotDb.createEmpty with correct parameters", async () => {
+      const fileName = "fileName";
+      const snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
+      const fake = sinon.fake.returns(snapshotDbMock.object);
+      sinon.replace(SnapshotDb, "createEmpty", fake);
+      sinon.stub(SnapshotConnection, "openFile");
 
-  it("builder calls provided callback function", async () => {
-    setupSnapshot();
-    const cb = sinon.spy();
+      await buildTestIModel(fileName, () => { });
 
-    await buildTestIModel("name", cb);
+      expect(fake.firstCall.firstArg).to.include(`${fileName}.bim`);
+      expect(fake.firstCall.lastArg).to.deep.equal({ rootSubject: { name: fileName } });
+    });
 
-    expect(cb.calledOnce);
-  });
+    it("builder calls provided callback function", async () => {
+      setupSnapshot();
+      const cb = sinon.spy();
 
-  it("builder saves database changes and closes it when callback succeeds", async () => {
-    setupSnapshot();
+      await buildTestIModel("name", cb);
 
-    await buildTestIModel("name", () => {});
+      expect(cb.calledOnce);
+    });
 
-    expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
-    expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
-  });
+    it("builder saves database changes and closes it when callback succeeds", async () => {
+      const snapshotDbMock = setupSnapshot();
 
-  it("builder saves database changes and closes it when callback throws", async () => {
-    setupSnapshot();
-    const cb = () => {
-      throw new Error("TestError");
-    };
+      await buildTestIModel("name", () => { });
 
-    try {
-      await buildTestIModel("name", cb);
-    } catch{}
+      expect(snapshotDbMock.verify((x) => x.saveChanges("Created test IModel"), moq.Times.once()));
+      expect(snapshotDbMock.verify((x) => x.close(), moq.Times.once()));
```

Comment:
You don't need to pass result of `verify` calls to `expect`. `verify` throws on failure, which causes the test to automatically fail.

Reply:
Removed `verify`